### ISSUE TITLE
DetailsList grid cells announce incorrect selection with Narrator

### ISSUE THREAD

**Issue by satbai** [OfficeDev/office-ui-fabric-react#6469]
### Bug Report
- __Package version(s)__: Fabric Core 9.6.0 and Fabric React 6.49.2
- __Browser and OS versions__: Edge + Narrator
 
#### Priorities and help requested:
Are you willing to submit a PR to fix? (Yes, No) No
Requested priority: (Blocking, High, Normal, Low) Normal
 
#### Describe the issue:
 When using the Narrator screen reader and navigating the DetailsList cells with the keyboard, Narrator announces incorrect selection state. When focusing on the whole row, the selection state is correct.
Repros in https://developer.microsoft.com/en-us/fabric#/components/detailslist

![image](https://user-images.githubusercontent.com/24945574/46040739-04b79d80-c0c6-11e8-9a79-19008f738950.png)
 
#### Actual behavior:

1. Turn on Narrator and Edge (optional: and Narratorbuddy) 
2. Navigate to https://developer.microsoft.com/en-us/fabric#/components/detailslist and Enable Modal Selection in the example
3. Select a row whose checkbox is selected (like in the example screen shot)
4. Use the Caps Lock + Right Arrow keys to navigate the grid cells
5. The individual cells will announce "non-selected" (see the screen shot)

#### Expected behavior:
 
5. If the row is selected, an individual cell should not announce "non-selected"

**Comment by natalieethell** [comment]
Hey @satbai thanks for opening up this issue! After investigating, it looks like this behavior is Narrator-specific. An issue was opened for it back in August 2017 [here](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/14844849/). 

> UIA Selection pattern is getting exposed by default on elements of role="grid". The SelectionItem Control Pattern should only be exposed on gridcell items when said items have the aria-selected attribute set to either true or false. Refer to: https://www.w3.org/TR/wai-aria-1.1/#grid

If you navigate through those grid examples on the [w3.org link](https://www.w3.org/TR/wai-aria-1.1/#grid), you'll see that Narrator even reads out "non-selected" for those gridcells in a grid that does not support selection.

One workaround I can think of is adding `aria-selected` to each gridcell. However, I'm not quite sure that representation would be correct, as the row is actually what is selected, not the individual gridcell. @KevinTCoughlin what do you think?

**Comment by natalieethell** [comment]
@KevinTCoughlin and I spoke offline and agree that adding aria-selected to each individual gridcell would not be the right move here. Since the row (the parent) is actually what is selected and not the gridcell (the child), it might be misleading to place aria-selected on the gridcell itself. 

Hopefully Narrator can fix the issue referenced above and we'll have that fixed as a result! 

In the meantime, @satbai let us know if you have any further questions.
